pkg/reconciler/instances/istio/reset/data/gatherer.go

```diff
@@ -83,34 +83,47 @@ func (i *DefaultGatherer) GetPodsWithDifferentImage(inputPodsList v1.PodList, im
 	return
 }
 
-func (i *DefaultGatherer) GetPodsWithoutSidecar(kubeClient kubernetes.Interface, retryOpts []retry.Option, sidecarInjectionEnabledbyDefault bool) (podsList v1.PodList, err error) {
+func (i *DefaultGatherer) GetPodsWithoutSidecar(kubeClient kubernetes.Interface, retryOpts []retry.Option, sidecarInjectionEnabledbyDefault bool) (podsList v1.PodList, labelWithWarningPodsList v1.PodList, err error) {
```

Comment:
Should it be part of this function? This returns the list of pods that should have a sidecar that currently do not have one.
Wouldn't it be cleaner and better to have another function to get all the pods without sidecar and label them after the proxy reset action?

Reply:
I moved this along with moving the labeling to separate action